components/turret.py

```diff
@@ -193,6 +200,11 @@ def execute(self) -> None:
 
         self.motor.set(ctre.ControlMode.MotionMagic, self.current_target_counts)
 
+        self.azimuth_history.appendleft(self.motor.getSelectedSensorPosition())
+        if len(self.azimuth_history) >= self.AZIMUTH_HISTORY_LEN:
+            # track the last second of azimuths
+            self.azimuth_history.pop()
```

Comment:
Instead of doing this yourself, would it be better to set a maximum length on the deque itself?